### ISSUE TITLE
REST call to query admin.user table where email equals a specific value

### ISSUE THREAD

**Issue by willishf** [supabase/supabase#29832]
Putting together a Java server side api that will use google to authenticate via oauth2 then see if that email already exists and if not create it. If it does exist get the row and using the uuid value see if that user has a specific role in public.user_roles that I created to support RLS. If it doesn't have the role then insert the proper role using the uuid

No matter what I try can not get the filtering to work on a specific value where it returns all rows. The docs don't have any example and tried what chatgpt and superbase AI suggested and no change. Did find a post from a couple years ago with someone having a very similar problem and the only way they could fix it was to change the table to text versus json. Details at the following https://community.draftbit.com/c/help/help-with-query-in-supabase-rest-api with how they solved it in the last post but not sure how the notation should change and if json or text makes a difference.

The following is the http request builder where I need the secret sauce after the ? where this returns all users. For the following two example I tried with email and encodedEmail in case the @ value was causing some problem. 

     String encodedEmail = URLEncoder.encode(email, StandardCharsets.UTF_8);
    HttpRequest request = HttpRequest.newBuilder()
            .uri(java.net.URI.create(SUPABASE_URL + "/auth/v1/admin/users?email=eq." + encodedEmail))
            .header("apikey", SUPABASE_SERVICE_ROLE_KEY)
            .header("Authorization", "Bearer " + SUPABASE_SERVICE_ROLE_KEY)
            .GET()
            .build();

In the process of looking for other similar posts, google Gemini suggested the following, which looks suspect that all values from the table will be returned, and then those values filtered, which isn't a nice thing to do to a very large table. 

            .uri(java.net.URI.create(SUPABASE_URL + "/auth/v1/admin/users?select=*,id&filter=email.eq." + encodedEmail))
            
Current behavior is that the query will return all rows and ignores any attempts to select for rows of a specific value which I assume are being used to build a SQL statement. 

Also very surprised of the lack of examples/discussion about how to do complex selection of specific rows

**Comment by willishf** [comment]
Totally missed that the REST endpoint /auth/v1/admin/users is for a procedure call versus the actual auth.users table. Was able to test the filter criteria on a public table and worked as expected. For oauth2 integration having the ability to query by user email for server side applications to get the user id that would allow populating roles etc would be helpful. Almost seems like the default use case should be forcing the user to have a search criteria and only return one row.

**Comment by willishf** [comment]
Not a bug as it relates to security but also not clear in the API docs. From a security standpoint does seem odd that even with appropriate level of access you can query all values but can't query based on filter criteria. An option should exist to query based on email to avoid workaround/hacks. For example I put a view together of the auth.users table for id, email and was able to query that table but now need to remember to lock it down from anon etc.

**Comment by anoosurf** [comment]
yes i'm having same issue .. i just want to get single row of user based on email .. using my secret admin key .. but i get the whole list of users .. which is so big and unnessary.. what is the solution ?

**Comment by willishf** [comment]
Turns out that call is to a procedure so no filtering. You can do your own
custom procedure as the recommended way to retrieve the data. I did a quick
fix and made a view with the two columns of interest of the users table in
the public schema. If you do this you need to lock it down so that anon has
no ability to do anything in the table or give a specific role access to
the select query. The recommended solution is your own custom procedure so
you don't create any additional security holes that someone could REST hack
your users.
